### PR TITLE
Non-fatal error in exported CMake function kokkos_check()

### PR DESCRIPTION
**Pull request by dalg24**
Address #2529

### CODE REVIEW

**Comment by dalg24** [comment]
> RESULT_VAR might be a better name? CMake tends to call similar things `OUTPUT_VAR`, etc.
> I can make this change, if you want.

Go ahead

File: cmake/KokkosConfig.cmake.in [hunk 1]
```diff
@@ -46,12 +46,12 @@ include(FindPackageHandleStandardArgs)
...
-#     [ARCH <archs>...]      # Target architectures (e.g. "Power9" and/or "Volta70")
-#     [OPTIONS <options>...] # Optional settings (e.g. "PROFILING")
-#     [TPLS <tpls>...]       # Third party libraries
-#     [RESULT <result>]      # Set a variable that indicates the result of the
-#                            # check instead of a fatal error
+#     [DEVICES <devices>...]   # Set of backends (e.g. "OpenMP" and/or "Cuda")
+#     [ARCH <archs>...]        # Target architectures (e.g. "Power9" and/or "Volta70")
+#     [OPTIONS <options>...]   # Optional settings (e.g. "PROFILING")
+#     [TPLS <tpls>...]         # Third party libraries
+#     [RETURN_VALUE <result>]  # Set a variable that indicates the result of the
```
Comment (dalg24) [inline cmake/KokkosConfig.cmake.in]: @jjwilke Why did you go with `RETURN_VALUE`?  Can you point to another function that use that naming convention?  I was trying to think of CMake functions and came across [`execute_process()`](https://cmake.org/cmake/help/latest/command/execute_process.html) that uses `RESULTS_VARIABLE`.

Comment (jjwilke) [reply cmake/KokkosConfig.cmake.in]: `exec_program` and `ctest_test` e.g.  The moral of the story is probably that CMake itself doesn't follow a consistent name.  Either is fine. Sorry, I didn't mean to bike shed.  I just wanted to be a bit clearer than `RESULT`.

File: cmake/KokkosConfig.cmake.in [hunk 2]
```diff
@@ -84,8 +86,10 @@ function(kokkos_check)
       set(KOKKOS_CHECK_SUCCESS FALSE)
     endif()
   endforeach()
-  if(NOT KOKKOS_CHECK_SUCCESS)
+  if(NOT KOKKOS_CHECK_SUCCESS AND NOT KOKKOS_CHECK_RESULT)
     message(FATAL_ERROR "Kokkos does NOT provide all backends and/or architectures requested")
+  else()
+    set(${KOKKOS_CHECK_RESULT} ${KOKKOS_CHECK_SUCCESS} PARENT_SCOPE)
```
Comment (dalg24) [inline cmake/KokkosConfig.cmake.in]: @jjwilke You missed that one.  OK to merge after this is fixed.

Comment (jjwilke) [reply cmake/KokkosConfig.cmake.in]: Correction. I missed *2*.

**Pull request merged**